File: src/com/crazicrafter1/lootcrates/sk/EvtLoot.java
package com.crazicrafter1.lootcrates.sk;

import ch.njol.skript.Skript;
import ch.njol.skript.lang.Literal;
import ch.njol.skript.lang.SkriptEvent;
import ch.njol.skript.lang.SkriptParser.ParseResult;
import ch.njol.skript.registrations.EventValues;
import ch.njol.skript.util.Getter;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;

public class EvtLoot extends SkriptEvent {

    static {
        Skript.registerEvent("Loot Execute", EvtLoot.class, SkriptLootEvent.class, "loot %string%");

        // Create the 'player' macro
        EventValues.registerEventValue(SkriptLootEvent.class, Player.class, new Getter<Player, SkriptLootEvent>() {
            @Override
            public Player get(SkriptLootEvent e) {
                return e.getPlayer();
            }
        }, 0);
    }

    private String tag = null;

    @Override
    public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
        tag = ((Literal<String>)args[0]).getSingle();
        return true;
    }

    @Override
    public boolean check(Event evt) {
        final SkriptLootEvent e = (SkriptLootEvent) evt;

        return e.getTag().equals(this.tag);
    }

    @Override
    public String toString(Event event, boolean b) {
        return "Loot execute event";
    }
}

File: src/main/java/com/crazicrafter1/lootcrates/sk/EvtLoot.kt
package com.crazicrafter1.lootcrates.sk

import ch.njol.skript.Skript
import ch.njol.skript.lang.Literal
import ch.njol.skript.lang.SkriptEvent
import ch.njol.skript.lang.SkriptParser
import ch.njol.skript.registrations.EventValues
import ch.njol.skript.util.Getter
import org.bukkit.entity.Player
import org.bukkit.event.Event

class EvtLoot : SkriptEvent() {
    private var tag: String? = null
    override fun init(args: Array<Literal<*>>, matchedPattern: Int, parseResult: SkriptParser.ParseResult): Boolean {
        tag = (args[0] as Literal<String?>).single
        return true
    }

    override fun check(evt: Event): Boolean {
        val e = evt as SkriptLootEvent
        return e.tag == tag
    }

    override fun toString(event: Event?, b: Boolean): String {
        return "Loot execute event"
    }

    companion object {
        init {
            Skript.registerEvent("Loot Execute", EvtLoot::class.java, SkriptLootEvent::class.java, "loot %string%")

            // Create the 'player' macro
            EventValues.registerEventValue(SkriptLootEvent::class.java, Player::class.java, object : Getter<Player?, SkriptLootEvent>() {
                override fun get(e: SkriptLootEvent): Player? {
                    return e.player
                }
            }, 0)
        }
    }
}
